tools/scoring/tests/dimensions/entity_identification_test.py

```diff
@@ -17,34 +17,178 @@
+                                  solution_path):
+    """Prepare argument for direct invocation of a dimension for purposes of
+    testing (i.e. mimic parse_config.py).
+
+    NOTE: this uses the simplified universe. If the test data references types
+    which are not contained therein, results will differ from those which use
+    the full universe because entities with missing types are skipped!
+
+      Arguments:
+        entity_type: the category of the dimension. (Literal[SIMPLE, COMPLEX])
+        proposed_path: the path to the proposed YAML file
+        solution_path: the path to the solution YAML file
+
+      Returns:
+        The appropriate value for the dimension's singular named argument"""
+
+    universe = BuildUniverse(use_simplified_universe=True)
+    proposed_config = validator.Deserialize([proposed_path])[0]
+    solution_config = validator.Deserialize([solution_path])[0]
+    deserialized_files = {PROPOSED: proposed_config, SOLUTION: solution_config}
+
+    deserialized_files_appended = ParseConfig._append_types(  # pylint: disable=protected-access
+        universe=universe,
+        deserialized_files=deserialized_files)
+
+    if entity_type == SIMPLE:
+      matches = ParseConfig._match_reporting_entities(  # pylint: disable=protected-access
+          proposed_entities=deserialized_files_appended[PROPOSED],
+          solution_entities=deserialized_files_appended[SOLUTION])
+
+      translations = ParseConfig._retrieve_reporting_translations(  # pylint: disable=protected-access
+          matches=matches,
+          proposed_entities=deserialized_files_appended[PROPOSED],
+          solution_entities=deserialized_files_appended[SOLUTION])
+
+      return translations
+    elif entity_type == COMPLEX:
+      return deserialized_files_appended
+
   def setUp(self):
     super().setUp()
-    # TODO: add real data (append cases to existing tests)
-    deserialized_files = {PROPOSED: {}, SOLUTION: {}}
-    self.entity_identification = EntityIdentification(
-        deserialized_files=deserialized_files).evaluate()
-
-  def testDirectlyAssignedAttributes(self):
-    self.assertEqual(self.entity_identification.correct_reporting, 0)
-    self.assertEqual(self.entity_identification.correct_ceiling_reporting, 0)
-    self.assertEqual(self.entity_identification.incorrect_reporting, 0)
-
-  def testInheritedCalculatedAttributes(self):
-    self.assertEqual(self.entity_identification.correct_total(), 0)
-    self.assertEqual(self.entity_identification.correct_ceiling(), 0)
-    self.assertEqual(self.entity_identification.incorrect_total(), 0)
-
-  def testInheritedResultProperties(self):
-    # These are `None` by virtue of the ceiling being falsy.
-    self.assertEqual(self.entity_identification.result_all, None)
-    self.assertEqual(self.entity_identification.result_reporting, None)
-    self.assertEqual(self.entity_identification.result_virtual, None)
+    all_features = 'tests/samples/virtual_entity.yaml'
```

Comment:
rename to `virtual_entity_file`? it is a file path, naming it all features is misleading here

Reply:
`_file` seems a bit redundant, particularly when it would be applied to everything in `/samples` for consistency. You're right about `all_features`, though — probably getting ahead of myself a bit.